### DISCUSSION TITLE
博士论文怎么让每级标题缩进，不缩进太难看了。

### DISCUSSION THREAD

**Discussion by earthmoon** [whutug/whu-thesis#103] [Q&A]
我看之前有人问了这个问题，答复是可以用在本科生论文上，我把那个办法用到博士论文，不可行。博士论文有办法实现缩进吗？

**Comment by sikouhjw** [comment]
https://github.com/whutug/whu-thesis/blob/a0cb5918638fece6c7fa171bb6be179dcf6a3a33/whu-thesis.cls#L263-L302

你需要的是设置 `indent` 选项，请阅读 `texdoc ctex`。

**Reply by earthmoon** [reply]
感觉有点难，等以后再看吧。

**Reply by sikouhjw** [reply]
照猫画虎
```tex
\ctexset{
  section = {
    indent = {2\ccwd},
  },
}
```

**Reply by earthmoon** [reply]
我把这段代码放到哪个位置呢？

**Reply by sikouhjw** [reply]
> 我把这段代码放到哪个位置呢？

放到导言区。

建议你先阅读 `texdoc lshort-zh` 再使用模板。

**Reply by earthmoon** [reply]
好。

**Reply by earthmoon** [reply]
你好，引用了一篇博士论文，第6页被引用。我使用的是\citep[6]{标签}   这种方式是对的吗？

**Reply by sikouhjw** [reply]
> 你好，引用了一篇博士论文，第6页被引用。我使用的是\citep[6]{标签} 这种方式是对的吗？

1. 一个 discussion 下专注一个话题；
2. 如果你使用的是 `bibtex`，参考文献样式由 `natbib` 宏包控制，请阅读手册 `texdoc natbib`。
   ![图片](https://user-images.githubusercontent.com/49770925/162438869-fbfa3be3-9800-4c8b-bf41-0c93e08ef590.png)

**Reply by earthmoon** [reply]
好的，谢谢，我先去把那本书打印下来吧。

**Reply by earthmoon** [reply]
书籍和博士论文的页码引用，好像满足不了国标哎。

**Reply by sikouhjw** [reply]
> 书籍和博士论文的页码引用，好像满足不了国标哎。

你说得对，我之前的答案是错的。

我不太了解 `bibtex` 能不能做到，但 `biblatex` 是可以的，参考 `texdoc biblatex-gb7714-2015`。
![图片](https://user-images.githubusercontent.com/49770925/162443631-147f4dbc-2426-482d-9048-ea2a80f8e208.png)

如何选择 `biblatex`，需要你去阅读手册。
https://github.com/whutug/whu-thesis/blob/a0cb5918638fece6c7fa171bb6be179dcf6a3a33/whu-thesis.cls#L837-L849

**Reply by earthmoon** [reply]
目前这个模板用的是bibtex对吧，感觉完全换了一个来处理文献，太有难度了。

**Reply by sikouhjw** [reply]
> 目前这个模板用的是bibtex对吧，感觉完全换了一个来处理文献，太有难度了。

模板默认是 `bibtex`，只需要设置一下 `style` 就可以切换到 `biblatex`。

1. 阅读手册，设置选项
2. 删除全部辅助文件
3. 再编译，注意用 `biber` 而不是 `bibtex` 编译参考文献部分

**Reply by earthmoon** [reply]
那vscode岂不是也要改，不然整个编译要点4次。你们会修改模板吗？我感觉我搞不定

**Reply by tanukihee** [reply]
我配置了 latexmk，你直接使用 latexmk 就行

**Reply by earthmoon** [reply]
> 我配置了 latexmk，你直接使用 latexmk 就行

具体的用法有没有教程呀。我对这个还是空白。

**Reply by tanukihee** [reply]
![image](https://user-images.githubusercontent.com/34062491/162447042-bf11c2e7-1712-4957-b7a3-87d03c311736.png)

选择带 latexmk 的两个之一就行

**Reply by sikouhjw** [reply]
https://github.com/whutug/whu-thesis/blob/a0cb5918638fece6c7fa171bb6be179dcf6a3a33/.vscode/settings.json#L5-L17

他在这里设置了，这两个都是调用 `latexmk`，也就是说你无需担心用 `bibtex` 还是 `biber` 编译。只需要在切换的时候删除辅助文件即可。

**Reply by earthmoon** [reply]
> 

原来这两个是这个意思。

**Reply by earthmoon** [reply]
> https://github.com/whutug/whu-thesis/blob/a0cb5918638fece6c7fa171bb6be179dcf6a3a33/.vscode/settings.json#L5-L17
> 
> 他在这里设置了，这两个都是调用 `latexmk`，也就是说你无需担心用 `bibtex` 还是 `biber` 编译。只需要在切换的时候删除辅助文件即可。

也就是说目前只需要设置biblatex，选择最上面两项编译就可以了对吧。

**Reply by tanukihee** [reply]
是

**Reply by earthmoon** [reply]
> 是

Thanks.

**Reply by tanukihee** [reply]
> 书籍和博士论文的页码引用，好像满足不了国标哎。

natbib问题

**Comment by earthmoon** [comment]
有人可以做一个biblatex适用国标的书籍页码引用的`style`吗？因为这个太不符合国标了，我看了《符合GB/T 7714-2015 标准的biblatex 参考文献样式》这个文档。由于对latex接触太少，学习的也少，确实目前是处理不来的。

**Reply by tanukihee** [reply]
无能为力，这个得去问 hushidong

**Reply by sikouhjw** [reply]
我在前面已经给出了详细的答案。
1. 选择 `bib-backend = biblatex`；
2. 选择 `type = doctor`
3. 使用 `\pagescite[150]{江泽民1989能源发展趋势及主要节能措施}` 或者 `\cite[150]{江泽民1989能源发展趋势及主要节能措施}`
4. ![图片](https://user-images.githubusercontent.com/49770925/162552052-d963ff1c-af98-4993-9b0f-6975ef9b0f17.png)

**Reply by earthmoon** [reply]
> 我在前面已经给出了详细的答案。
> 
> 1. 选择 `bib-backend = biblatex`；
> 2. 选择 `type = doctor`
> 3. 使用 `\pagescite[150]{江泽民1989能源发展趋势及主要节能措施}` 或者 `\cite[150]{江泽民1989能源发展趋势及主要节能措施}`
> 4. ![图片](https://user-images.githubusercontent.com/49770925/162552052-d963ff1c-af98-4993-9b0f-6975ef9b0f17.png)

你好，因为我水平菜，昨天的没看懂。麻烦了。
1. 我修改成这样：`\msg_new:nnn { whu-thesis } { cite-style 无效 }
  {
    bib-backend = biblatex 下，\\`
2.`type = doctor`这个在哪呢？我没有搜索到
3.还可以使用 \citep 和 \citet 吗，因为两种格式引用都需要
谢谢

**Reply by sikouhjw** [reply]
1. `type` 加到 `\documentclass` 里，即 `\documentclass[type=doctor,class=design]{whu-thesis}`
2. `bib-backend = biblatex` 加到 `\whusetup{style/bib-backend}` 里，即
   ```tex
   style              =
   {
     graphics-path  = {{figures/}{data/}},
     list-of-figures,
     list-of-tables,
     bib-backend    = biblatex,
   },
   ```
3. 根据 `texdoc biblatex-gb7714-2015`
   > 其它命令 `\citet`、`\citep`、`\citetns`、`\citepns`、`\upcite`、`\inlinecite` 分别等价于 `\authornumcite`、`\cite`、`\textcite`、`\parencite`、`\cite`、`\parencite`。

   所以你可以使用更容易输入的 `\citet`、`\citep`。

**Reply by tanukihee** [reply]
默认就是博士所以不用加（

**Reply by tanukihee** [reply]
> > 我在前面已经给出了详细的答案。
> > 
> > 1. 选择 `bib-backend = biblatex`；
> > 2. 选择 `type = doctor`
> > 3. 使用 `\pagescite[150]{江泽民1989能源发展趋势及主要节能措施}` 或者 `\cite[150]{江泽民1989能源发展趋势及主要节能措施}`
> > 4. ![图片](https://user-images.githubusercontent.com/49770925/162552052-d963ff1c-af98-4993-9b0f-6975ef9b0f17.png)
> 
> 你好，因为我水平菜，昨天的没看懂。麻烦了。
> 1. 我修改成这样：`\msg_new:nnn { whu-thesis } { cite-style 无效 }
>   {
>     bib-backend = biblatex 下，\\`
> 2.`type = doctor`这个在哪呢？我没有搜索到
> 3.还可以使用 \citep 和 \citet 吗，因为两种格式引用都需要
> 谢谢



（你真的读了文档吗

**Reply by earthmoon** [reply]
> 1. `type` 加到 `\documentclass` 里，即 `\documentclass[type=doctor,class=design]{whu-thesis}`
> 2. `bib-backend = biblatex` 加到 `\whusetup{style/bib-backend}` 里，即
>    ```
>    style              =
>    {
>      graphics-path  = {{figures/}{data/}},
>      list-of-figures,
>      list-of-tables,
>      bib-backend    = biblatex,
>    },
>    ```
> 3. 根据 `texdoc biblatex-gb7714-2015`
>    > 其它命令 `\citet`、`\citep`、`\citetns`、`\citepns`、`\upcite`、`\inlinecite` 分别等价于 `\authornumcite`、`\cite`、`\textcite`、`\parencite`、`\cite`、`\parencite`。
>    
>    
>    所以你可以使用更容易输入的 `\citet`、`\citep`。

你好，编译器使用xalatexmk，文章里的文献引用直接显示成粗体标签了，现在正在用lualatexmk编译，lualatexmk比xalatexmk慢很多。

**Reply by tanukihee** [reply]
改bib后端需要删除辅助文件

**Reply by earthmoon** [reply]
> 改bib后端需要删除辅助文件

删除辅助文件后，编译器使用xalatexmk，文章内的引用还是显示为粗体引用标签，是我哪里搞错了么。

**Reply by earthmoon** [reply]
> > > 我在前面已经给出了详细的答案。
> > > 
> > > 1. 选择 `bib-backend = biblatex`；
> > > 2. 选择 `type = doctor`
> > > 3. 使用 `\pagescite[150]{江泽民1989能源发展趋势及主要节能措施}` 或者 `\cite[150]{江泽民1989能源发展趋势及主要节能措施}`
> > > 4. ![图片](https://user-images.githubusercontent.com/49770925/162552052-d963ff1c-af98-4993-9b0f-6975ef9b0f17.png)
> > 
> > 
> > 你好，因为我水平菜，昨天的没看懂。麻烦了。
> > 
> > 1. 我修改成这样：`\msg_new:nnn { whu-thesis } { cite-style 无效 }   { bib-backend = biblatex 下，\\`
> >    2.`type = doctor`这个在哪呢？我没有搜索到
> >    3.还可以使用 \citep 和 \citet 吗，因为两种格式引用都需要
> >    谢谢
> 
> （你真的读了文档吗

似懂非懂地读了，对这个软件没有系统的认识，所以操作有时候是凭感觉。

**Reply by earthmoon** [reply]
> 

报错有这三个：
[链接](https://bthc-my.sharepoint.com/:i:/g/personal/mr_iem_im/EWM2gjCh4JNHhSKWVH0SdpIBnzWkQG8kRo2-GGJ3sDE04A?e=FQZ3Wp)

**Reply by earthmoon** [reply]
> > 
> 
> 报错有这三个： [链接](https://bthc-my.sharepoint.com/:i:/g/personal/mr_iem_im/EWM2gjCh4JNHhSKWVH0SdpIBnzWkQG8kRo2-GGJ3sDE04A?e=FQZ3Wp)

我把vccode重启了，然后删除了附属文件，现在没有报错，只是文内引用的文献就是粗体引用的标签。

**Reply by sikouhjw** [reply]
看起来你用的依然是 `bibtex`，没有用 `biber`。

建议去看 `texdoc install-latex` 学下命令行编译。

**Reply by earthmoon** [reply]
latexmk不是可以用biber吗，我看昨天的回复是可以的。
我用texworks编译，第一步用xelatex没问题，第二步用biber报错：
INFO - This is Biber 2.17
INFO - Logfile is 'whu-thesis-demo.blg'
INFO - Reading 'whu-thesis-demo.bcf'
INFO - Using all citekeys in bib section 0
INFO - Processing section 0
INFO - Looking for bibtex file 'ref/refs' for section 0
ERROR - Cannot find 'ref/refs'!
INFO - ERRORS: 1
不是biber相关的问题吧

**Reply by sikouhjw** [reply]
> latexmk不是可以用biber吗，我看昨天的回复是可以的。 我用texworks编译，第一步用xelatex没问题，第二步用biber报错： INFO - This is Biber 2.17 INFO - Logfile is 'whu-thesis-demo.blg' INFO - Reading 'whu-thesis-demo.bcf' INFO - Using all citekeys in bib section 0 INFO - Processing section 0 INFO - Looking for bibtex file 'ref/refs' for section 0 ERROR - Cannot find 'ref/refs'! INFO - ERRORS: 1 不是biber相关的问题吧

哦……`biber` 要在后面加 `.bib`，也就是改成 `ref/refs.bib`。

**Reply by earthmoon** [reply]
> > latexmk不是可以用biber吗，我看昨天的回复是可以的。 我用texworks编译，第一步用xelatex没问题，第二步用biber报错： INFO - This is Biber 2.17 INFO - Logfile is 'whu-thesis-demo.blg' INFO - Reading 'whu-thesis-demo.bcf' INFO - Using all citekeys in bib section 0 INFO - Processing section 0 INFO - Looking for bibtex file 'ref/refs' for section 0 ERROR - Cannot find 'ref/refs'! INFO - ERRORS: 1 不是biber相关的问题吧
> 
> 哦……`biber` 要在后面加 `.bib`，也就是改成 `ref/refs.bib`。

对了，加bib就对了。比如我引用的页码是8-15，我输入\citep[8-15]{标签} ，虽然输出正确，但是这个` -` 显示偏下。这个`-`的输入有什么讲究吗？在这里严谨的输入应该是什么呀

**Reply by sikouhjw** [reply]
> 但是这个` -` 显示偏下。

字体决定的，别纠结了。

> 这个`-`的输入有什么讲究吗？在这里严谨的输入应该是什么呀

页码之间的连接，用 `--`。参考《如何正确使用和打出英文的破折号？》[^1]。

[^1]: https://www.zhihu.com/question/19715623

**Reply by earthmoon** [reply]
> > 但是这个` -` 显示偏下。
> 
> 字体决定的，别纠结了。
> 
> > 这个`-`的输入有什么讲究吗？在这里严谨的输入应该是什么呀
> 
> 页码之间的连接，用 `--`。参考《如何正确使用和打出英文的破折号？》[1](#user-content-fn-1-71576a6f7a6ffc6dc63afd261bcd9029)。
> 
> ## Footnotes
> 1. https://www.zhihu.com/question/19715623 [↩](#user-content-fnref-1-71576a6f7a6ffc6dc63afd261bcd9029)

好的，谢谢你！这个问题总算解决了。

**Reply by earthmoon** [reply]
> > 但是这个` -` 显示偏下。
> 
> 字体决定的，别纠结了。
> 
> > 这个`-`的输入有什么讲究吗？在这里严谨的输入应该是什么呀
> 
> 页码之间的连接，用 `--`。参考《如何正确使用和打出英文的破折号？》[1](#user-content-fn-1-71576a6f7a6ffc6dc63afd261bcd9029)。
> 
> ## Footnotes
> 1. https://www.zhihu.com/question/19715623 [↩](#user-content-fnref-1-71576a6f7a6ffc6dc63afd261bcd9029)

我在I-short cn文档上面查到了，确实是这样。谢谢
